src/pages/FilesPage/FilesTabs/UploadDatasourceModal/UploadDatasourceModal.tsx

```diff
@@ -23,6 +29,11 @@ export const UploadDatasourceModal: FC<UploadDatasourceModalProps> = function Up
 
   const canCloseModal = !isUploading;
 
+  const onComplete = () => {
+    clearFilesList();
```

Comment:
Just for you information. There was another option - it is possible to set 'destroy modal after close'
`destroyOnClose`
https://3x.ant.design/components/modal/